lms/static/js/financial-assistance/views/financial_assistance_form_view.js

```diff
@@ -101,6 +102,21 @@
 
                 setExtraData: function(data) {
                     return _.extend(data, this.user_details);
+                },
+
+                validateCountry: function() {
```

Comment:
You need to reload the page for your new country to be passed through.  So yes, we rely on the view being rerendered